common/transform/src/test/java/ai/h2o/mojos/deploy/common/transform/MojoScorerTest.java

```diff
@@ -136,35 +134,326 @@ void verifyScoreRequestWithOriginalShapley_fails() {
+    // Given
+    System.setProperty("shapley.enable", "true");
+
+    ScoreRequest request = new ScoreRequest();
+    request.addFieldsItem("field1");
+    request.addRowsItem(toRow("text"));
+    request.setRequestShapleyValueType(ShapleyType.TRANSFORMED);
+    MojoFrame dummyMojoFrame = generateDummyTransformedMojoFrame();
+    given(scoreRequestConverter.apply(any(), any()))
+        .willReturn(dummyMojoFrame);
+    ScoreResponse dummyResponse = generateDummyResponse();
+    given(scoreResponseConverter.apply(any(), any()))
+        .willReturn(dummyResponse);
+
+    MojoScorer scorer = dummyScorer();
+
+    // When
+    scorer.score(request);
+
+    // Then all ok
+  }
+
+  @Test
+  void verifyScoreRequestWithOriginalShapley_ShapleyEnabled_Succeeds() {
+    // Given
+    System.setProperty("shapley.enable", "true");
+
+    ScoreRequest request = new ScoreRequest();
+    request.addFieldsItem("field1");
+    request.addRowsItem(toRow("text"));
+    request.setRequestShapleyValueType(ShapleyType.ORIGINAL);
+    MojoFrame dummyMojoFrame = generateDummyTransformedMojoFrame();
+    given(scoreRequestConverter.apply(any(), any()))
+        .willReturn(dummyMojoFrame);
+    ScoreResponse dummyResponse = generateDummyResponse();
+    given(scoreResponseConverter.apply(any(), any()))
+        .willReturn(dummyResponse);
+
+    MojoScorer scorer = dummyScorer();
+
+    // When
+    scorer.score(request);
+
+    // Then all ok
+  }
+
+  @Test
+  void verifyScoreRequestWithoutShapley_ShapleyOptionAll_Succeeds() {
+    // Given
+    System.setProperty("shapley.types.enabled", "ALL");
+
+    ScoreRequest request = new ScoreRequest();
+    request.addFieldsItem("field1");
+    request.addRowsItem(toRow("text"));
+    MojoFrame dummyMojoFrame = generateDummyTransformedMojoFrame();
+    given(scoreRequestConverter.apply(any(), any()))
+        .willReturn(dummyMojoFrame);
+    ScoreResponse dummyResponse = generateDummyResponse();
+    given(scoreResponseConverter.apply(any(), any()))
+        .willReturn(dummyResponse);
+
+    MojoScorer scorer = dummyScorer();
+
+    // When
+    scorer.score(request);
+
+    // Then all ok
+  }
+
+  @Test
+  void verifyScoreRequestWithTransformedShapley_ShapleyOptionAll_Succeeds() {
+    // Given
+    System.setProperty("shapley.types.enabled", "ALL");
+
+    ScoreRequest request = new ScoreRequest();
+    request.addFieldsItem("field1");
+    request.addRowsItem(toRow("text"));
+    request.setRequestShapleyValueType(ShapleyType.TRANSFORMED);
+    MojoFrame dummyMojoFrame = generateDummyTransformedMojoFrame();
+    given(scoreRequestConverter.apply(any(), any()))
+        .willReturn(dummyMojoFrame);
+    ScoreResponse dummyResponse = generateDummyResponse();
+    given(scoreResponseConverter.apply(any(), any()))
+        .willReturn(dummyResponse);
+
+    MojoScorer scorer = dummyScorer();
+
+    // When
+    scorer.score(request);
+
+    // Then all ok
```

Comment:
In the specific-shapley tests, do we need to test that the scorer is indeed only recognizing/returning what we want?

Due to time constraint, I confess I am not looking very closely at how the test class props are set.  I'm not sure if your props/mocks allow making these verifications.

Reply:
I think its fair for now... TBH I'm mostly stealing from tests that already preexisted and just extending for additional cases. Eventually it would probably make sense to have more indepth tests but we also consume the exception if there is one. So I'm inclined to say this is good enough for now.

Reply:
We could mock the response out of the method to actually return some expected value I suppose.

Reply:
So afaik we cannot mock private methods... and the public exposed stuffs doesn't really provide the ability for us to check anything else except that it doesn't barf... So really unless I dive deeper under the hood this is probably as good as it gets.